### PR TITLE
group the release info

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

this makes the codename globally accessible in the repo. The release is
not only for NixOS anymore.

continuation on #48122

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: lib/trivial.nix [hunk 1]
```diff
@@ -92,11 +92,16 @@ rec {
...
+  release-info = import ../release-info.nix;
+
   # The current full nixpkgs version number.
   version = release + versionSuffix;
 
   # The current nixpkgs version number as string.
-  release = lib.strings.fileContents ../.version;
+  release = release-info.release;
+
+  # Note: the first letter is bumped on every release.  It's an animal.
```
Comment (samueldr) [inline lib/trivial.nix]: The note might be more relevant at the location the value is defined.

File: lib/trivial.nix [hunk 2]
```diff
@@ -92,11 +92,16 @@ rec {
...
+
   # The current full nixpkgs version number.
   version = release + versionSuffix;
 
   # The current nixpkgs version number as string.
-  release = lib.strings.fileContents ../.version;
+  release = release-info.release;
+
+  # The current nixpkgs version code name.
+  codeName = release-info.codename;
```
Comment (edolstra) [inline lib/trivial.nix]: Why not just put the release info here? I don't really see the point in putting it in a separate file (except for easy parseability, but that's already lost if you use a Nix expression instead of a plain file like .version).

Comment (zimbatm) [reply lib/trivial.nix]: In terms of data locality, the release and code name get updated at the same time so it's best to keep them together.

In terms of discoverability, `lib/trivial.nix` is a misc util library. I don't have a strong opinion where the information should be except that `lib/trivial.nix` doesn't seem like the right place. I can see myself running `cat release-info.nix` on a channel or checkout to quickly figure out where I stand.

In terms of parse-ability, Nix is not too hard to parse. JSON would be even easier but doesn't allow comments.

Comment (zimbatm) [reply lib/trivial.nix]: fixed in 49a9011

File: lib/trivial.nix (hunk 1 above)
Comment (zimbatm) [reply lib/trivial.nix]: fixed in 49a9011

**Comment by zimbatm** [comment]
simplified the PR to the minimum, hope that won't be controversial.

**Review by infinisil — APPROVED** [review]
